### ISSUE TITLE
Text to speech (tts) not included in MsgType enum 

### ISSUE THREAD

**Issue by martijnvanderwoud** [messagebird/java-rest-api#26]
The `com.messagebird.objects.VerifyRequest` type has a `type` attribute, which must be of type `com.messagebird.objects.MsgType`. Although text to speech ("tts") is a valid option for verification requests, this type is not included in the `MsgType` enum.  I guess this value should be added to the `MsgType` enum. Correct?

**Comment by samwierema** [comment]
@martijnvanderwoud thanks for opening an issue. 

It seems like a bit of an oversight on our part. `MsgType` is not necessarily fully suited for this purpose as some options there can't be passed to Verify (e.g. `premium`). What we should ideally do is have a `VerifyMessageType` enum that list all the possible options and use that. However, that might be a breaking change. I'll discuss with my colleagues how to approach this best.

Of course if you have suggestion or want to contribute a change, please feel welcome to do so.

**Comment by martijnvanderwoud** [comment]
Hi Sam, thanks for the quick response. I think we should look for a solution that does not break backwards compatibility. How about adding `tts` to `MsgType`, and throwing a `RunTimeException` (something like `UnsupportedMessageTypeException`) when someone tries to set a `MsgType` that is not available? We could also add a `static boolean supports(MsgType messageType)` method to the `VerifyRequest` class in case people want to determine at runtime which options are available (for populating a dropdown of available message types, for instance).

If you would like I can create a pull request for this, but let's first agree on how to approach this.

**Comment by martijnvanderwoud** [comment]
Hi Sam, I was wondering if you already have decided how you want to approach this.

**Comment by martijnvanderwoud** [comment]
Still curious how you want to solve this @samwierema :)

**Comment by siovene** [comment]
Hi, I am affected by this issue. Is there any possible workaround to get a TTS Verify using the Java API? Perhaps a way to do a raw request?

**Comment by martijnvanderwoud** [comment]
@siovene we worked around the issue by creating our own payload class: 
```
public class VerificationRequest {

	private String recipient;

	private String originator;

	private String type;

	private String template;

	private Integer timeout;

	private String voice;

	private String language;

	VerificationRequest(String recipient) {
		this.recipient = recipient;
	}

       ... getters and setters ....
}
```

Then sending a verification request is a matter of 
```
response = messageBirdService.sendPayLoad("/verify", verificationRequest, Verify.class);
```

**Comment by epels** [comment]
Thanks @martijnvanderwoud for sharing your solution - that does indeed same to be the best workaround for now.

I'm planning a new major release of this client that fixes this and other issues. As @samwierema mentioned, that will be a breaking change. When TTS verifications become available out-of-the-box, I will update this issue. Thanks!

CC @siovene

**Comment by siovene** [comment]
Thanks @martijnvanderwoud, the workaround worked for me. Thanks @epels for the update.

**Comment by epels** [comment]
@martijnvanderwoud @siovene Thanks for your patience - this feature is now available in [`v2.0.0`](https://mvnrepository.com/artifact/com.messagebird/messagebird-api/2.0.0).

```java
VerifyRequest verifyRequest = new VerifyRequest("31612345678");
verifyRequest.setType(VerifyType.TTS);
        
Verify verify = messageBirdClient.sendVerifyToken(verifyRequest);
        
// ...
```

In case you have any further questions, please feel free to re-open this issue or create a new one.
